File: fs/src/main/scala/farjs/fs/popups/FoldersHistoryController.scala
package farjs.fs.popups

import scommons.react._

object FoldersHistoryController extends FunctionComponent[FoldersHistoryControllerProps] {

  private[popups] var foldersHistoryPopup: ReactClass = FoldersHistoryPopup

  protected def render(compProps: Props): ReactElement = {
    val props = compProps.plain

    if (props.showPopup) {
      <(foldersHistoryPopup)(^.plain := FoldersHistoryPopupProps(
        onChangeDir = { dir =>
          props.onClose()
          props.onChangeDir(dir)
        },
        onClose = props.onClose
      ))()
    }
    else null
  }
}

File: fs/popups/FoldersHistoryController.mjs
import React from "react";
import FoldersHistoryPopup from "./FoldersHistoryPopup.mjs";

const h = React.createElement;

/**
 * @typedef {{
 *  readonly showPopup: boolean;
 *  onChangeDir(dir: string): void;
 *  onClose(): void;
 * }} FoldersHistoryControllerProps
 */

/**
 * @param {FoldersHistoryControllerProps} props
 */
const FoldersHistoryController = (props) => {
  const { foldersHistoryPopup } = FoldersHistoryController;

  if (props.showPopup) {
    return h(foldersHistoryPopup, {
      onChangeDir: (dir) => {
        props.onClose();
        props.onChangeDir(dir);
      },
      onClose: props.onClose,
    });
  }

  return null;
};

FoldersHistoryController.displayName = "FoldersHistoryController";
FoldersHistoryController.foldersHistoryPopup = FoldersHistoryPopup;

export default FoldersHistoryController;
